File: t/VB/OcrModulesDemo/UI/MyItemData.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text

Friend Structure MyItemData
   Private _zoneType As Integer
   Private _characterFilters As Integer
   Private _selectedDocFormatIndex As Integer
   Private _selectedOcrModuleIndex As Integer
   Private _name As String
   Private _originalName As String
   Private _imageFileName As String

   Public Sub New(ByVal name_Renamed As String, ByVal originalName_Renamed As String, ByVal zoneType As Integer, ByVal characterFilters As Integer, ByVal imageFileName_Renamed As String, ByVal selectedDocFormatIndex As Integer, ByVal selectedOcrModuleIndex_Renamed As Integer)
      _name = name_Renamed
      _originalName = originalName_Renamed
      _zoneType = zoneType
      _characterFilters = characterFilters
      _imageFileName = imageFileName_Renamed
      _selectedDocFormatIndex = selectedDocFormatIndex
      _selectedOcrModuleIndex = selectedOcrModuleIndex_Renamed
   End Sub

   Public Property ZoneType() As Integer
      Get
         Return _zoneType
      End Get
      Set(ByVal value As Integer)
         _zoneType = value
      End Set
   End Property

   Public Property CharacterFilters() As Integer
      Get
         Return _characterFilters
      End Get
      Set(ByVal value As Integer)
         _characterFilters = value
      End Set
   End Property

   Public Property Name() As String
      Get
         Return _name
      End Get
      Set(ByVal value As String)
         _name = value
      End Set
   End Property

   Public Property OriginalName() As String
      Get
         Return _originalName
      End Get
      Set(ByVal value As String)
         _originalName = value
      End Set
   End Property

   Public Property SelectedDocumentFormatIndex() As Integer
      Get
         Return _selectedDocFormatIndex
      End Get
      Set(ByVal value As Integer)
         _selectedDocFormatIndex = value
      End Set
   End Property

   Public Property SelectedOcrModuleIndex() As Integer
      Get
         Return _selectedOcrModuleIndex
      End Get
      Set(ByVal value As Integer)
         _selectedOcrModuleIndex = value
      End Set
   End Property

   Public Property ImageFileName() As String
      Get
         Return _imageFileName
      End Get
      Set(ByVal value As String)
         _imageFileName = value
      End Set
   End Property

   Public Overrides Function ToString() As String
      Return _name
   End Function
End Structure

File: t/CS/OcrModulesDemo/UI/MyItemData.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Text;

namespace OcrModulesDemo
{
   struct MyItemData
   {
      int _zoneType;
      int _characterFilters;
      int _selectedDocFormatIndex;
      int _selectedOcrModuleIndex;
      string _name;
      string _originalName;
      string _imageFileName;

      public MyItemData(string name, string originalName, int zoneType, int characterFilters, string imageFileName, int selectedDocFormatIndex, int selectedOcrModuleIndex)

      {
         _name = name;
         _originalName = originalName;
         _zoneType = zoneType;
         _characterFilters = characterFilters;

         _imageFileName = imageFileName;
         _selectedDocFormatIndex = selectedDocFormatIndex;
         _selectedOcrModuleIndex = selectedOcrModuleIndex;
      }

      public int ZoneType
      {
         get
         {
            return _zoneType;
         }
         set
         {
            _zoneType = value;
         }
      }

      public int CharacterFilters
      {
         get
         {
            return _characterFilters;
         }
         set
         {
            _characterFilters = value;
         }
      }

      public string Name
      {
         get
         {
            return _name;
         }
         set
         {
            _name = value;
         }
      }

      public string OriginalName
      {
         get
         {
            return _originalName;
         }
         set
         {
            _originalName = value;
         }
      }

      public int SelectedDocumentFormatIndex
      {
         get
         {
            return _selectedDocFormatIndex;
         }
         set
         {
            _selectedDocFormatIndex = value;
         }
      }

      public int SelectedOcrModuleIndex
      {
         get
         {
            return _selectedOcrModuleIndex;
         }
         set
         {
            _selectedOcrModuleIndex = value;
         }
      }

      public string ImageFileName
      {
         get
         {
            return _imageFileName;
         }
         set
         {
            _imageFileName = value;
         }
      }

      public override string ToString()
      {
         return _name;
      }
   }
}
